client/signup/main.jsx

```diff
@@ -124,11 +123,6 @@ const Signup = React.createClass( {
 		} );
 		recordSignupStart();
 
-		// Signup updates the cart through `SignupCart`. To prevent
-		// synchronization issues and unnecessary polling, the cart is disabled
-		// here.
-		disableCart();
```

Comment:
:+1: That's how I understood it too. Disable just turns off polling until we dispatch for a cart again. Cart is cleared on the back-end if successful. Not sure how to handle stale carts, apart from clearing them every time for a `signup` cart.